### PR TITLE
Upgrade cadl compiler and cadl-ranch to the latest

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/1604

Upgrade to cadl compiler 0.36.0
- How to handle `never` type in cadl https://github.com/Azure/autorest.typescript/pull/1614/files#diff-840cacc995c77ae70376fe1bcc625c368a54d96d77dc4a5ff07de45c129a5619?

Integrate cadl ranch:
- How to enable auth token testing with message https://dev.azure.com/azure-sdk/public/_build/results?buildId=1926490&view=logs&j=ea067a9b-50f7-5bd8-5a42-5349a4ef9f90&t=f3a82242-5b86-51ad-918b-f63213dfd42a&l=483 ?
```
  1) AuthOauth2Client Rest Client
       should return 204 when the token is valid:
     AssertionError: Error: Bearer token authentication is not permitted for non-TLS protected (non-https) URLs.
```

==> Handled in a customized policy

### CODE REVIEW

File: packages/cadl-typescript/src/transform/transformAnnotationDetails.ts [hunk 1]
```diff
@@ -12,16 +12,20 @@ export function transformAnnotationDetails(
   program: Program
 ): AnnotationDetails | undefined {
   // Extract paged metadata from Azure.Core.Page
+  let annotationDetails = {
```
Comment (qiaozha) [inline packages/cadl-typescript/src/transform/transformAnnotationDetails.ts:15]: Is there any where changing this annotationDetails ? why not using const ？

File: packages/cadl-typescript/test/commands/cadl-ranch-list.ts [hunk 2]
```diff
@@ -16,10 +16,11 @@ export const cadls: CadlRanchConfig[] = [
...
   },
-  {
-    outputPath: "dictionary",
-    inputPath: "dictionary"
-  },
+  // TODO: comment this out once the dict issue resolved
+  // {
+  //   outputPath: "dictionary",
+  //   inputPath: "dictionary"
+  // },
```
Comment (qiaozha) [inline packages/cadl-typescript/test/commands/cadl-ranch-list.ts:23]: what's the issue about dictionary?

**Review by qiaozha — APPROVED** [review]

File: packages/cadl-typescript/test/commands/cadl-ranch-list.ts (hunk 2 above)
Comment (MaryGao) [reply packages/cadl-typescript/test/commands/cadl-ranch-list.ts:23]: It reports error when generation the response.ts I will fix it in next pr.

```
SyntaxError: ',' expected. (2:26)
  1 | import { HttpResponse } from "@azure-rest/core-client";
> 2 | import { InnerModelOutput> } from "./outputModels";
    |                          ^
  3 |
  4 | /** The request has succeeded. */
prettier\index.js:8837:18)
```

File: packages/cadl-typescript/src/modelUtils.ts [hunk 3]
```diff
@@ -670,7 +674,7 @@ function mapCadlIntrinsicModelToTypeScript(
           schema.valueTypeName = valueType.name;
           if (usage && usage.includes(SchemaContext.Output)) {
             schema.outputTypeName = `Record<string, ${valueType.name}Output>`;
-            schema.outputValueTypeName = `${valueType.name}Output>`;
+            schema.outputValueTypeName = `${valueType.name}Output`;
```
Comment (MaryGao) [inline packages/cadl-typescript/src/modelUtils.ts:677]: Remove the extra character `>`

File: packages/cadl-typescript/test/integration/authOauth2.spec.ts [hunk 4]
```diff
@@ -2,24 +2,42 @@ import Outh2ClientFactory, {
...
-    );
+      }
+    });
+    client.pipeline.removePolicy({
+      name: bearerTokenAuthenticationPolicyName
+    });
+    client.pipeline.addPolicy(policy);
   });
 
-  it.skip("should return 204 when the token is valid", async () => {
```
Comment (MaryGao) [inline packages/cadl-typescript/test/integration/authOauth2.spec.ts:22]: Re-enable the token auth testing

File: packages/cadl-rlc-test/test/confidentialLedger/cadl-output/src/clientDefinitions.ts [hunk 5]
```diff
@@ -88,54 +88,47 @@ export interface ConfidentialLedgerOperations {
     | ConfidentialLedgerCreateLedgerEntry201Response
     | ConfidentialLedgerCreateLedgerEntryDefaultResponse
   >;
-  /** Get a LedgerEntry */
```
Comment (qiaozha) [inline packages/cadl-rlc-test/test/confidentialLedger/cadl-output/src/clientDefinitions.ts:91]: why this is removed.

**Review by qiaozha — APPROVED** [review]

**Pull request merged**